.github/workflows/ci.yml

```diff
@@ -156,16 +155,32 @@ jobs:
           key: nginx-kubernetes-gateway-${{ github.run_id }}-${{ github.run_number }}
       - name: Docker Buildx
         uses: docker/setup-buildx-action@v1
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        if: ${{ startsWith(github.ref, 'refs/tags') }}
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: |
+            name=ghcr.io/nginxinc/nginx-kubernetes-gateway
+          tags: |
+            type=semver,pattern={{version}}
       - name: Build Image Debian
         uses: docker/build-push-action@v2
         with:
           file: build/Dockerfile
           context: '.'
           target: local
-          load: true
+          tags: ${{ steps.meta.outputs.tags }}
```

Comment:
You also need to update the tag on line 192 with `${{ steps.meta.outputs.version }}`

Reply:
That seems like a good idea but I've just been making the changes the team has asked me to make.  As of now, the image on line 192 is being tagged with needs.vars.outputs.sha on the main branch pipelines.  @pleshakov Do you want me to update that so the trivy image is tagged with the release version on release branches?

Reply:
> As of now, the image on line 192 is being tagged with needs.vars.outputs.sha on the main branch pipelines

I'm not sure what you mean? You removed the tag with the `sha` and added the tags from the metadata action. With these changes now nothing is creating an image tagged with the sha

Reply:
@pammecrandall we'd like the trivy scan to be done both on the main branch and also on release branches.

Reply:
```      - name: Run Trivy vulnerability scanner
        uses: aquasecurity/trivy-action@0.2.3
        continue-on-error: true
        with:
          image-ref: nginx/nginx-kubernetes-gateway:${{ needs.vars.outputs.sha }}
          format: 'template'
          template: '@/contrib/sarif.tpl'
          output: 'trivy-results-nginx-kubernetes-gateway.sarif'```

Reply:
@pleshakov 

> we'd like the trivy scan to be done both on the main branch and also on release branches.

Okay.  That sounds good.  I'll make that update now.

Reply:
Yeah what I'm saying is that nothing creates this build 
```
image-ref: nginx/nginx-kubernetes-gateway:${{ needs.vars.outputs.sha }}
```
so the scan will fail

Reply:
hmm.  This just came from the main branch and it seemed to be working.  It's getting late here so I'll take a look in the morning.

Reply:
I see what you're saying. I changed this so it's just using one image and the ${{ steps.meta.outputs.version }} tag you suggested.  Thank you!  This works in my fork but it looks like the code has some lint errors causing this to fail.